### DISCUSSION TITLE
emqx metrics(bytes) is inaccurate?

### DISCUSSION THREAD

**Discussion by Gypsying** [emqx/emqx#9989] [General]
emqx [bytes metrics](https://www.emqx.io/docs/zh/v4.4/advanced/metrics-and-stats.html#metrics-stats) is inaccurate?

i used emqx cluster(15 node deployed at vms), the bytes metrics pushed to pushgateway is much less than the vm's network interface flux : emqx stats(emqx_bytes_received/emqx_bytes_sent) is 1MB while device network interface is 25MB...

so, what's mean the emqx_bytes_received and emqx_bytes_sent?

**Comment by Gypsying** [comment]
any help?

**Comment by Gypsying** [comment]
...

**Reply by hjianbo** [reply]
Hi, @Gypsying Sorry for the late response here.

The `emqx_bytes_received/sent` only accumulated in the MQTT protocol layer (i.e from 1883, 8883 ports). Not counting traffic used by HTTP services (i.e: used by Dashboard static files, HTTP APIs)

**Reply by Gypsying** [reply]
> Hi, @Gypsying Sorry for the late response here.
> 
> The `emqx_bytes_received/sent` only accumulated in the MQTT protocol layer (i.e from 1883, 8883 ports). Not counting traffic used by HTTP services (i.e: used by Dashboard static files, HTTP APIs)

well, how to get the total byte received/sent used by emqx ?

**Reply by kjellwinblad** [reply]
If you are interested in monitoring all network traffic going in and out of EMQX, I don't believe there is anything built-in for that purpose. However, there are several Linux utilities available to help with this task. Two examples you can explore are `iftop` and `nethogs`.

**Reply by Gypsying** [reply]
> If you are interested in monitoring all network traffic going in and out of EMQX, I don't believe there is anything built-in for that purpose. However, there are several Linux utilities available to help with this task. Two examples you can explore are `iftop` and `nethogs`.

Okay, thanks for replies, 
i will use the node_exporter to monitor the device-dimension traffic finally...
